smart-contract-extensions/contracts/KeyPurchaser.sol

```diff
@@ -0,0 +1,162 @@
+pragma solidity 0.5.16;
+
+import 'hardlydifficult-ethereum-contracts/contracts/lifecycle/Stoppable.sol';
+import '@openzeppelin/upgrades/contracts/Initializable.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import 'unlock-abi-1-3/IPublicLockV6.sol';
+
+/**
+ * @notice Purchase a key priced in any ERC-20 token - either once or as a regular subscription.
+ * This allows the user to purchase or subscribe to a key with 1 tx (`approve`)
+ * or if the token supports it, with 1 signed message (`permit`).
+ *
+ * The user can remove approval to cancel anytime.
```

Comment:
Just to be clear, this is referring to `erc-20 approve()` right?

Reply:
yup. I could improve the comment

Reply:
updated